metagpt/tools/search_engine_googleapi.py

```diff
@@ -25,38 +26,11 @@
 
 
 class GoogleAPIWrapper(BaseModel):
-    google_api_key: Optional[str] = None
-    google_cse_id: Optional[str] = None
+    google_api_key: str = os.environ.get("GOOGLE_API_KEY")
+    google_cse_id: str = os.environ.get("GOOGLE_CSE_ID")
     loop: Optional[asyncio.AbstractEventLoop] = None
     executor: Optional[futures.Executor] = None
 
-    class Config:
-        arbitrary_types_allowed = True
-
-    @validator("google_api_key", always=True)
-    @classmethod
-    def check_google_api_key(cls, val: str):
-        val = val or CONFIG.google_api_key
-        if not val:
-            raise ValueError(
-                "To use, make sure you provide the google_api_key when constructing an object. Alternatively, "
-                "ensure that the environment variable GOOGLE_API_KEY is set with your API key. You can obtain "
-                "an API key from https://console.cloud.google.com/apis/credentials."
-            )
-        return val
-
-    @validator("google_cse_id", always=True)
-    @classmethod
-    def check_google_cse_id(cls, val: str):
-        val = val or CONFIG.google_cse_id
```

Comment:
When google_api_key or google_cse_id is not set, the validator will provide guidance. If delete this code and there are no google_api_key or google_cse_id, it will raise an exception, making it difficult for users to determine the cause of the error from the error message.

Reply:
Thank you for the valuable feedback on my revised code. I appreciate your insights and the time you took to review my work.